src/cli.ts

```diff
@@ -25,200 +23,123 @@
+       */
+      if (!options.quietExitCode) {
+        for (const result of Object.values(results)) {
+          if (result.status === 'failed') {
+            process.exit(1);
+          }
         }
-      });
-    } else {
-      addSuite(absPath);
-    }
-  }
-  return suites.values();
-}
-
-(async () => {
-  /**
-   * Transform `.ts` files out of the box by invoking
-   * the `ts-node` via `transpile-only` mode that compiles
-   * TS files without doing any extensive type checks.
-   *
-   * We must register `ts-node` _before_ loading inline
-   * scripts too because otherwise we will not be able to
-   * require `.ts` configuration files.
-   */
-  /* eslint-disable-next-line @typescript-eslint/no-var-requires */
-  require('ts-node').register({
-    transpileOnly: true,
-    compilerOptions: {
-      esModuleInterop: true,
-      allowJs: true,
-      target: 'es2020',
-    },
-  });
-
-  if (options.inline) {
-    const source = await readStdin();
-    loadInlineScript(source);
-  } else {
-    /**
-     * Preload modules before running the suites
-     */
-    const modules = [].concat(options.require || []).filter(Boolean);
-    for (const name of modules) {
-      if (isDepInstalled(name)) {
-        require(name);
-      } else {
-        throw new Error(`cannot find module '${name}'`);
       }
+    } catch (e) {
+      console.error(e);
+      process.exit(1);
     }
-    /**
-     * Handle piped files by reading the STDIN
-     * ex: ls example/suites/*.js | npx @elastic/synthetics
-     */
-    const files =
-      program.args.length > 0
-        ? program.args
-        : (await readStdin()).split('\n').filter(Boolean);
-    const suites = await prepareSuites(files);
-    requireSuites(suites);
-  }
-
-  /**
-   * Use the NODE_ENV variable to control the environment
-   */
-  const environment = process.env['NODE_ENV'] || 'development';
-  /**
-   * Validate and handle configs
-   */
-  const config =
-    options.config || !options.inline
-      ? readConfig(environment, options.config)
-      : {};
-  const params = merge(config.params, options.params || {});
-
-  /**
-   * Favor playwright options passed via cli to inline playwright options
-   */
-  const playwrightOptions = merge.all([
-    config.playwrightOptions || {},
-    options.playwrightOptions || {},
-    {
-      headless: options.headless,
-      chromiumSandbox: options.sandbox,
-      ignoreHTTPSErrors: options.ignoreHttpsErrors,
-    },
-  ]);
+  });
 
-  const results = await run({
-    params: Object.freeze(params),
-    networkConditions: options.throttling
-      ? parseNetworkConditions(options.throttling as string)
-      : undefined,
-    environment,
-    playwrightOptions,
-    ...options,
+program
+  .command('push')
+  .description(
+    'Push monitors to create new montors with Kibana monitor management UI'
+  )
+  .action(() => {
+    throw new Error('TODO: Implement me');
   });
 
-  if (!options.quietExitCode) {
```

Comment:
I'm OK with this being removed it was causing a significant issue in creating this PR, but it is a nice feature.

On some CI systems this would make sense with JUnit XML output since you might want the test command to always succeed, but base the pass/fail nature of the CI task on a subsequent step that parses that output.

Reply:
I thought we introduced this feature only for HB, Thanks for the explanation Andrew. I can bring it back.

Reply:
Done.